File: Advance Python/Oops/tempCodeRunnerFile.py

class A:
    def __init__(self):
        print("A constructor")

    def aInstanceMethod(self):
        print("a instance method")
        
    @classmethod
    def aClassMethod(cls):
        print("a class method")

    @staticmethod
    def aStaticMethod():
        print("a static method")
        
        
class B(A):
    def __init__(self):
        super().__init__()
        print("B constructor")

    def aInstanceMethod(self):
        super().aInstanceMethod()
        print("B instance method")
        
    @classmethod
    def aClassMethod(cls):
        super().aClassMethod()
        print("B class method")

    @staticmethod
    def aStaticMethod():
        A.aStaticMethod()
        print("B static method")
        
        
b = B()
print("Calling b methods")
b.aInstanceMethod()
B.aClassMethod()
B.aStaticMethod()

File: frontend/JS/Oops/tempCodeRunnerFile.js


// polymorphism
class A{

    constructor(){
        console.log("A constructor");
    }

    m1(){
        console.log("A class implementation");
    }
}

class B extends A{

    constructor(){
        new A();
        console.log("B constructor");
    }

    m1(){
        console.log("B class implementation");
        super.m1()
    }
}


let b = new B();
b.m1();


